x/alloc/keeper/msg_server_fund_fee_collector.go

```diff
@@ -0,0 +1,29 @@
+package keeper
+
+import (
+	"context"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/public-awesome/stargaze/v5/x/alloc/types"
+)
+
+func (k msgServer) FundFairburnPool(goCtx context.Context, msg *types.MsgFundFairburnPool) (*types.MsgFundFairburnPoolResponse, error) {
+	ctx := sdk.UnwrapSDKContext(goCtx)
+
+	sender, err := sdk.AccAddressFromBech32(msg.Sender)
+	if err != nil {
+		return nil, err
+	}
+	err = k.FundFairburlPool(ctx, sender, msg.Amount)
+	if err != nil {
+		return nil, err
+	}
+	ctx.EventManager().EmitEvent(
```

Comment:
There's also a Fair Burn event on the contract side. Maybe we should get rid of that one and keep this. Then there'll no longer be a need to pass in a mutable `Response`.

Reply:
this only does the funding part and the burn side still part of the contracts we could probably do both things here like a single message that does burn and fund.

Reply:
Oh that's right. Either way is fine. Since this already works can just leave it as is.